### PR TITLE
[Project A][MacSilicon][BaskarGopinath] Upgrade to Java 17

### PR DESCRIPTION
**Pull request by baskargopinath**
- OS used : MacOS Sonoma 14.5 on M2 Macbook Pro
- JDK distribution used : Azul zulu-17 version 17.0.11
- IDE used : IntelliJ IDEA 2022.3.2 (Community Edition)
- Link to the uploaded JAR file : https://github.com/baskargopinath/AB3-J17/releases/tag/J17
- No problems faced

### CODE REVIEW

**Comment by hjungwoo01** [comment]
Tested on Mac Silicon with JDK-FX 17.0.11.fx-zulu, and working.

Steps I followed:
1. Downloaded the JAR file from the PR.
2. Switched to JDK-FX 17.0.11.fx-zulu using SDKMAN.
3. Ran the JAR file using `java -jar MacSillicon-BaskarGopinath.jar`.
4. Performed a smoke test to check basic functionality.

Results:
Everything is functioning as expected.

Screenshots:
<p align="left">
  Screenshot 2024-07-04 at 3 46 09 PM
Screenshot 2024-07-04 at 3 48 44 PM
</p>

**Comment by itstrueitstrueitsrealitsreal** [comment]
Tested on Mac Silicon with JDK-FX 17.0.11.fx-zulu, and working.

### Steps followed:
1. Downloaded the JAR file from the PR.
2. Switched to JDK-FX 17.0.11.fx-zulu using SDKMAN.
3. Ran the JAR file using `java -jar MacSillicon-BaskarGopinath.jar`.
4. Performed a smoke test to verify basic functionality.

### Results:
Everything is functioning as expected.

### Screenshots:
Screenshot 2024-07-04 at 4 53 47 PM

**Comment by jyue487** [comment]
Tested on Mac Silicon with JDK-FX 17.0.11.fx-zulu, and working.

Steps and results are identical to above.

**Comment by teoks0199** [comment]
Tested on Windows 11, and working.

Steps I followed:

1. Downloaded the JAR file from the PR.
2. Ran the JAR file using java -jar MacSillicon-BaskarGopinath.jar.
3. Performed a smoke test to check basic functionality.

image

**Comment by macareonie** [comment]
Tested on Windows 11 with Oracle OpenJDK version 17.0.11, and working.

image

**Comment by bennyLCK** [comment]
Smoke-Tested on Windows 11, using Oracle OpenJDK version 17.0.10 on Windows Powershell Terminal same as @teoks0199, basic commands were run to ensure full functionality of the application.

image

**Comment by baskargopinath** [comment]
Tested on Mac Silicon with JDK-FX 17.0.11.fx-zulu, and working.

Steps:
Downloaded the JAR file from the PR.
Switched to JDK-FX 17.0.11.fx-zulu
Ran the JAR file using java -jar MacSillicon-BaskarGopinath.jar.
Performed a smoke test to check basic functionality.

Results:
Everything is functioning as expected.

Screenshots:
image

**Comment by aureliony** [comment]
Tested on Ubuntu 22.04.4 LTS with zulu jdk 17.0.11. JAR is working fine.

Steps followed:
Downloaded the JAR file from the release.
Ran the JAR file using `java -jar MacSillicon-BaskarGopinath.jar`.
Performed a smoke test to check basic functionality.

Results:
Everything is functioning as expected.

Screenshot from 2024-07-06 19-34-43

**Comment by gongg21** [comment]
Tested on **MacSilicon** with JDK-FX 17.0.11.fx-zulu, and working fine.
Switched using SDKMAN, ensure it is **17.0.11.fx-zulu**, note the `.fx` suffix.
Add/Delete command tested, works.
Help command tested, works.
Ran JAR file using `java -jar <filename>`.

Screenshot 2024-07-07 at 5 30 13 PM

**Comment by Carlintyj** [comment]
Tested on Mac (Silicon) using JDK-FX 17.0.11.fx-zulu and working.

Downloaded the JAR release, ran the jar file using java -jar BaskarGopinath.jar
Performed smoke test to check functionality.

Screenshot 2024-07-07 at 7 03 57 PM

**Comment by drustanyjt** [comment]
Tested on Linux (Ubuntu 22.04) using OpenJDK 17.0.11. Working.

Steps taken:

1. Downloaded release
2. Ran jar file using `java -jar MacSillicon-BaskarGopinath.jar`
3. Tested with basic commands

image

**Comment by brein62** [comment]
Tested on **Windows 10** using JDK-FX 17.0.11.fx-zulu, and working.

## Procedure

Ran the JAR file on PowerShell using `java -jar MacSillicon-BaskarGopinath.jar` and performed a smoke test by running several commands to ensure basic functionality of the application:

- `add n/John p/12345678 e/johnd@example.com a/123 Example Road t/friends`: works
- `find Roy`: works
- `list`: works
- `help`: works
- `delete 3`: works
- `clear`: works

## Results
The JAR functions as expected and all commands run during the smoke test work properly.

## Screenshots
MacSilicon-BaskarGopinath

**Comment by jedkohjk** [comment]
Tested on Windows 11 and working.

Steps I followed:
1. Downloaded the JAR file from the release.
2. Ran `java -jar MacSillicon-BaskarGopinath.jar`.
3. Tested some basic commands.

Screenshot

File: build.gradle [hunk 1]
```diff
@@ -5,11 +5,12 @@ plugins {
     id 'application'
     id 'jacoco'
 }
+application {
+    mainClass = 'seedu.address.Main'
+}
```
Comment (damithc) [inline build.gradle:10]: @baskargopinath Some of these changes are not strictly needed for the Java 17 migration. If you think they are still useful, you can send separate PRs for them, with the justification.

**Comment by damithc** [comment]
Closing, as done in https://github.com/nus-oss/AB3-J17/commit/fe805d8553bbc329644371a6e732a8317ea6686f
